### DISCUSSION TITLE
Exclude Dto Properties

### DISCUSSION THREAD

**Discussion by muratyuceer** [thepirat000/Audit.NET#500] [Q&A]
Hello, I want to exclude some properties of dto class when logging. Like a password field etc..
How can I do that with this lib.?
Thanks

**Comment by thepirat000** [comment]
Please provide more context. What are you auditing? Which extensions are you using?

**Reply by muratyuceer** [reply]
Hi, I auditing for webapi. I need to put attribute to properties. I want say to json deserilaze ignore it just when you loging

**Reply by thepirat000** [reply]
Do you target .NET >= 5? System.Text.Json or Newtonsoft.Json?

**Reply by muratyuceer** [reply]
I using .net 6, but i dont want to ignore that property from everywhere just when it loging

**Comment by thepirat000** [comment]
Assuming you use the default serialization mechanism for .NET 6 `System.Text.Json`.

An option is to create a custom Json Converter that ignores the properties, and then register the converter to the `JsonSettings` exposed by `Audit.NET`.

For example, suppose you have a `UserData` class and you want to ignore its `Password` property when serializing:

```c#
public class UserData
{
    public string Name { get; set; }
    public string Password { get; set; } 
}
```

You can implement a custom JSON converter as follows:

```c#
public class UserDataConverter : JsonConverter<UserData>
{
    public override void Write(Utf8JsonWriter writer, UserData value, JsonSerializerOptions options)
    {
        writer.WriteStartObject();

        using (var document = JsonDocument.Parse(JsonSerializer.Serialize(value)))
        {
            foreach (var property in document.RootElement.EnumerateObject())
            {
                if (property.Name != "Password")
                {
                    property.WriteTo(writer);
                }
                    
            }
        }

        writer.WriteEndObject();
    }

    public override UserData Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        throw new NotImplementedException();
    }
}
```

And in your startup:

```c#
   Audit.Core.Configuration.JsonSettings.Converters.Add(new UserDataConverter());
```

So any `UserData` serialization made by Audit.NET library, will ignore the property, but it will be included in any other case
